File: sample-app-java/app/src/main/java/com/bluejeans/android/sdksample/participantlist/WaitingRoomParticipantListAdapter.java
package com.bluejeans.android.sdksample.participantlist;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.bluejeans.android.sdksample.R;
import com.bluejeans.bluejeanssdk.meeting.ParticipantsService;
import java.util.List;
import org.jetbrains.annotations.NotNull;

public class WaitingRoomParticipantListAdapter extends
        RecyclerView.Adapter<WaitingRoomParticipantListAdapter.ViewHolder> {

    private List<ParticipantsService.Participant> participants;
    private IWaitingRoomActionsCallback waitingRoomActionsCallback;

    public WaitingRoomParticipantListAdapter(List<ParticipantsService.Participant> participants,
                                             IWaitingRoomActionsCallback waitingRoomActionsCallback) {
        this.participants = participants;
        this.waitingRoomActionsCallback = waitingRoomActionsCallback;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.waiting_room_dialog_row, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        ParticipantsService.Participant participant = participants.get(position);

        holder.tvWaitingRoomParticipantName.setText(participant.getName());

        holder.btnAdmit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                waitingRoomActionsCallback.admitParticipant(participant);
            }
        });

        holder.btnDeny.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                waitingRoomActionsCallback.denyParticipant(participant);
            }
        });
    }

    @Override
    public int getItemCount() {
        return this.participants.size();
    }

    public interface IWaitingRoomActionsCallback {
        void admitParticipant(ParticipantsService.Participant participant);
        void denyParticipant(ParticipantsService.Participant participant);
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        @NotNull
        private final TextView tvWaitingRoomParticipantName;
        @NotNull
        private final Button btnDeny;
        @NotNull
        private final Button btnAdmit;
        
        public ViewHolder(@NotNull View view) {
            super(view);
            tvWaitingRoomParticipantName = view.findViewById(R.id.tv_waiting_room_participant_name);
            btnDeny = view.findViewById(R.id.btn_deny);
            btnAdmit = view.findViewById(R.id.btn_admit);
        }
    }
}

File: sample-app-kotlin/app/src/main/java/com/bluejeans/android/sdksample/participantlist/WaitingRoomParticipantListAdapter.kt
/**
 * Copyright (c) 2021 Blue Jeans Network, Inc. All rights reserved.
 */
package com.bluejeans.android.sdksample.participantlist

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bluejeans.android.sdksample.R
import com.bluejeans.bluejeanssdk.meeting.ParticipantsService

class WaitingRoomParticipantListAdapter(
    private val participants: List<ParticipantsService.Participant>,
    private val waitingRoomActionCallback: IWaitingRoomActionsCallback
) : RecyclerView.Adapter<WaitingRoomParticipantListAdapter.ViewHolder>() {

    interface IWaitingRoomActionsCallback {
        fun admitParticipant(participant: ParticipantsService.Participant)
        fun denyParticipant(participant: ParticipantsService.Participant)
    }

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val tvWaitingRoomParticipantName: TextView = view.findViewById(R.id.tv_waiting_room_participant_name)
        val btnDeny: Button = view.findViewById(R.id.btn_deny)
        val btnAdmit: Button = view.findViewById(R.id.btn_admit)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.waiting_room_dialog_row, parent, false)

        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val participant = participants.elementAt(position)
        holder.tvWaitingRoomParticipantName.text = participant.name

        holder.btnAdmit.setOnClickListener {
            waitingRoomActionCallback.admitParticipant(participant)
        }

        holder.btnDeny.setOnClickListener {
            waitingRoomActionCallback.denyParticipant(participant)
        }
    }

    override fun getItemCount(): Int {
        return participants.size
    }
}
